### ISSUE TITLE
Something strange occurred when training and evaluating

### ISSUE THREAD

**Issue by Jasonsun1993** [experiencor/keras-yolo3#33]
I just trained my dataset for 50 epochs, but when it almost completed, it stopped early. And it couldn't be evaluated when I ran evaluate.py. Like this:
![2](https://user-images.githubusercontent.com/33592598/39279125-be4729e2-4929-11e8-88f3-7b160cfd6215.JPG)
![3](https://user-images.githubusercontent.com/33592598/39279130-c5a274bc-4929-11e8-9121-d67c66fe8cec.JPG)
Can anybody tell me what the problem is and how to solve it?
@experiencor

**Comment by Jasonsun1993** [comment]
And why the model was not compiled??

**Comment by Jasonsun1993** [comment]
![131](https://user-images.githubusercontent.com/33592598/39282745-56f0c2e4-493d-11e8-9a5c-67c83d58d482.jpg)

**Comment by shensheng27** [comment]
you might check your val data label , xmin xmax ymin ymax could be pixel index or pixel index / image height or width.
try print the detection result and label result

**Comment by Jasonsun1993** [comment]
@shensheng27 I used the labelImg tool to make my dataset. And the xmin ymin xmax ymax should be relative positions of the image. Besides, the author said we could use labelImg tool to make xml file #25

**Comment by shensheng27** [comment]
it's ok to using labellmg because it save annotaion in pascal voc xml.
Did you mean predict.py is ok to draw correct box but evaluate.py fail to calculate map? I could not see too much difference between two file.
I would suggest just draw both detection and label in result pic and see where is wrong.
Or do print in 84line of utils.py

**Comment by Jasonsun1993** [comment]
@shensheng27 I mean the test results (predict.py file) are good with bouding boxes and AP scores (the object can be correctly detected). But I got zero mAP after running evaluate.py. I guess that there is something wrong with calculating mAP. Do you know in which file the mAP calculating process is??

**Comment by shensheng27** [comment]
keras-yolo3/utils/utils.py   do print in 84line  would be helpful
try pycharm or other IDE...

**Comment by Jasonsun1993** [comment]
@shensheng27 Hi, there is another problem. The predict.py cannot read the video which is MP4 format. But I check the code, it shows that it can detect video, right? Did you ever test video?

**Comment by experiencor** [comment]
@shensheng27 Yes, it does. I make the videos in the readme using predict.py on mp4.

**Comment by Jasonsun1993** [comment]
@experiencor but it is so strange, when I ran the predict.py, I found that the code couldn't read my mp4 format file. The images could be read correctly without any error tip. Why did it happen?

**Comment by yuyijie1995** [comment]
@Jasonsun1993  Hello，Can you tell me about how to use the evaluate?
![image](https://user-images.githubusercontent.com/30818917/41088302-c1b9c0b2-6a71-11e8-8025-0d34d6e08d0b.png)
I used my testdata to evaluate but the code can not extract the raw image.
![image](https://user-images.githubusercontent.com/30818917/41088367-f0530316-6a71-11e8-9dd8-3e9df67a4fbb.png)
This is my config doc,can you help me figure out what is wrong about it? Thank you very much!

**Comment by experiencor** [comment]
@yuyijie1995 Try to order your labels alphabetically. There's a bug but I have yet to fix.

**Comment by yuyijie1995** [comment]
Thanks a lot @experiencor

**Comment by yuyijie1995** [comment]
@experiencor  Can I just remove the .pkl file and the code with it?Because I founded that the .pkl file was something wrong always.
